tcpclient/connection_test.go

```diff
@@ -0,0 +1,41 @@
+package tcpclient
+
+import (
+	"testing"
+	"time"
+)
+
+func TestGetConnectionStatus(t *testing.T) {
+	connectionDescription := Connection{
+		ID:      0,
+		status:  ConnectionDialing,
+		metrics: connectionMetrics{},
+	}
+	if connectionDescription.GetConnectionStatus() != ConnectionDialing {
+		t.Error("GetConnectionStatus is not working")
+	}
+}
+
+func TestGetTCPProcessingDurationEstablished(t *testing.T) {
+
+	connectionDescription := Connection{
+		ID:      0,
+		status:  ConnectionEstablished,
+		metrics: connectionMetrics{tcpEstablishedDuration: 1 * time.Second},
+	}
+	if connectionDescription.GetTCPProcessingDuration(ConnectionEstablished) != 1*time.Second {
+		t.Error("GetTCPProcessingDuration is not working")
+	}
+}
+
+func TestGetTCPProcessingDurationErrored(t *testing.T) {
```

Comment:
Instead of writing tests for getters (that are too simple to worth a test, from my perspective), I'd invite you to write the test for the String function given that, at least, there's a basic logic. In fact, writing the test you may realize we are not printing something equivalent to "Connection %d has become %s after %s" for Errors, and maybe its worth (and maybe implementing this is just simpler consolidating the "processing" time into one (insisting on an old idea :) )

Reply:
ok, TODO

Reply:
done in commit 8e08125

Reply:
:+1: